### PR TITLE
feat: Add delete ticket functionality

### LINKED ISSUE #359: Add delete ticket option
Deleting a ticket is much more basic than creating one, so add this functionality first

### PR DESCRIPTION
**Pull request by iamareebjamal**
- Remove automatic attendee download on getting tickets
- Add delete method in event service, ticket repository and presenter
- Use data binding to show intermediate state of ticket deleting

Fixes #359

screenshot-1500955963775

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal
what if the ticket is booked by the attendees?

**Comment by iamareebjamal** [comment]
That should be handled by the server once the order API is complete. For the app, the constraints make the attendee get deleted as well, but as soon as the event detail or attendee page is opened, it is reloaded from the server. 

In short, the server can have detached attendees as of now, but we must implement basic client CRUD

**Comment by mejariamol** [comment]
@iamareebjamal let's take Niranjan's view on this

@niranjan94 do we want this feature in the app? Ticket deletion.

**Comment by niranjan94** [comment]
I don't think ticket deletion is to be done at this point. Because, this still needs a lot to think through in terms of business logic. Some of the main things are:
- Handling/triggering refunds
- Cancellation charges (and charges based on when they're cancelling)
- What if it was paid in cash at the venue ? How do we handle refunds.
- Does deleted tickets open up instantly for new booking or is there a wait period

**Comment by niranjan94** [comment]
So, have the app code if you want. But hide the delete button for now.

**Comment by mejariamol** [comment]
>Does deleted tickets open up instantly for new booking or is there a wait period

@iamareebjamal @niranjan94 Some confusion there.
I think this feature is about deleting the ticket type of the event. right? @iamareebjamal 
@niranjan94 might have misunderstood it with individual attendee deletion.

**Comment by niranjan94** [comment]
@mejariamol updated my comment. Yes. I had mis-understood. But my original concerns still remain.

File: app/src/main/java/org/fossasia/openevent/app/common/contract/Pipe.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.fossasia.openevent.app.common.contract;
+
+public interface Pipe<T> {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/contract/Pipe.java]: where is this used?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/contract/Pipe.java]: To push the ticket from viewholder to adapter. It is essentially a Java 8 or Rx consumer, but Java 8 one does not work on Android and Rx one needs to throw exception.

**Comment by iamareebjamal** [comment]
@mejariamol Please review this

File: app/src/main/java/org/fossasia/openevent/app/common/data/models/Ticket.java [hunk 2]
```diff
@@ -42,6 +43,8 @@
     public int minOrder;
     public String salesStartsAt;
 
+    public final ObservableBoolean deleting = new ObservableBoolean();
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/data/models/Ticket.java]: shouldn't it be `isDeleting`?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/data/models/Ticket.java]: boolean fields should be deleting and getters/setters be isDeleting setDeleting

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by niranjan94** [comment]
@iamareebjamal please resolve conflicts here :)

**Comment by iamareebjamal** [comment]
@niranjan94 Done :smile:

**Pull request merged**